lib/seoshop-api/client/order.rb

```diff
@@ -0,0 +1,67 @@
+module Seoshop
+  class Client::Order
+    class CheckoutError < StandardError
+    end
+
+    attr_reader :order_id, :client, :checkout_details, :payment_details, :shipping_details, :checkout_id,
+                :checkout_body, :validation_body, :number
+
+    def initialize(client, checkout_details, payment_details, shipping_details)
+      @client = client
+      @payment_details = payment_details
+      @shipping_details = shipping_details
+      @checkout_response = @client.post("#{language}/checkouts.json", checkout_details)
+      @checkout_body ||= @checkout_response.body
+      if @checkout_body
+        @checkout_id ||= @checkout_body['id']
+      end
+    end
+
+    def valid?
+      update_shipping_and_paymeny_info
+      @validation_body = client.get("#{language}/checkouts/#{checkout_id}/validate.json").body
+      @validation_body[:validated]
+    end
+
+    def save!
+      response = client.post("#{language}/checkouts/#{checkout_id}/order.json", checkout_attrs)
+      if response.status == 201 || response.status == 200
+        @order_id = response.body['order_id']
+      else
+        error = response.body[:error]
+        fail CheckoutError, "Checkout with id: #{checkout_id} could not convert into an order. Errors: #{error[:message]} (code: #{error[:code]})"
+      end
+    end
+
+    def update!(attrs)
+      response = client.put("#{language}/orders/#{order_id}.json", { order: attrs })
+      unless response.status == 200
```

Comment:
Instead of only checking `response.status == 200` you can do `response.success?`